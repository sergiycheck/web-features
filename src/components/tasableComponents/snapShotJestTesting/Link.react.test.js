import React from "react";
import renderer from "react-test-renderer";
import CustomLink from "./Link.react";

test("Link changes the class when hovered", () => {
  const component = renderer.create(
    <CustomLink page="https://example.com">Webpage</CustomLink>
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  //trigger the callback
  tree.props.onMouseEnter();

  //re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  //trigger the callback
  tree.props.onMouseLeave();

  //re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
