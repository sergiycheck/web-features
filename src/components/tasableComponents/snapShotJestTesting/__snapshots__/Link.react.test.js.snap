// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link changes the class when hovered 1`] = `
<a
  className="normal"
  href="https://example.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Webpage
</a>
`;

exports[`Link changes the class when hovered 2`] = `
<a
  className="hovered"
  href="https://example.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Webpage
</a>
`;

exports[`Link changes the class when hovered 3`] = `
<a
  className="normal"
  href="https://example.com"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
>
  Webpage
</a>
`;
